### PR TITLE
Support smart_variables as a Hash or Array

### PR DESCRIPTION
**Pull request by starchow**
Currently, the smart_variables must be an SQL query. Sometimes I know the exactly values and I want write it as a Hash or Array.

I also updated README to easier understand how to use linked_columns, smart_columns, smart_variables.

Please check out my pull request and merge it if it good enough. 
Thanks!

### CODE REVIEW

**Comment by ankane** [comment]
Thanks @starchow!  I like the idea.  A few thoughts:

- not sure it makes sense to call humanize on hash keys
- not a huge fan of the readme change (not very DRY)

**Comment by starchow** [comment]
Hi @ankane,

If smart_variable is a Hash, the key will be display to user (So it should be readable), and the value is for query. 
I think the current README is hard to understand how to use smart_columns, smart_variables, linked_columns. After 2 months, I found how to use it. But it's up to you, I will revert the README if you don't want to merge it.

Thanks!

**Comment by evandrodp** [comment]
+1

**Comment by starchow** [comment]
I updated my pull request, please double check @ankane!

File: app/controllers/blazer/queries_controller.rb [hunk 1]
```diff
@@ -53,7 +53,11 @@ def show
       data_source = Blazer.data_sources[@query.data_source]
       @bind_vars.each do |var|
         query = data_source.smart_variables[var]
-        if query
+        if query.is_a? Hash
+          @smart_vars[var] = query.map {|k,v| [k,v]}
+        elsif query.is_a? Array
+          @smart_vars[var] = query
```
Comment (ankane) [inline app/controllers/blazer/queries_controller.rb]: I think this should be `query.map { |v| [v, v] }`.

File: app/controllers/blazer/queries_controller.rb [hunk 2]
```diff
@@ -53,7 +53,11 @@ def show
       data_source = Blazer.data_sources[@query.data_source]
       @bind_vars.each do |var|
         query = data_source.smart_variables[var]
-        if query
+        if query.is_a? Hash
+          @smart_vars[var] = query.map {|k,v| [k,v]}
```
Comment (ankane) [inline app/controllers/blazer/queries_controller.rb]: Please preserve the existing style where possible - `query.map { |k,v| [k, v] }`.

File: app/controllers/blazer/dashboards_controller.rb [hunk 3]
```diff
@@ -36,7 +36,11 @@ def show
       @bind_vars.each do |var|
         @data_sources.each do |data_source|
           query = data_source.smart_variables[var]
-          if query
```
Comment (ankane) [inline app/controllers/blazer/dashboards_controller.rb]: Can we DRY this with the queries controller?

File: README.md [hunk 4]
```diff
@@ -631,4 +631,4 @@ That’s awesome! Here are a few ways you can help:
 - Write, clarify, or fix documentation
 - Suggest or add new features
 
-Check out the [dev app](https://github.com/ankane/blazer-dev) to get started.
+Check out the [dev app](https://github.com/ankane/blazer-dev) to get started.
```
Comment (ankane) [inline README.md]: What's this change?

File: lib/generators/blazer/templates/config.yml [hunk 5]
```diff
@@ -23,6 +23,10 @@ data_sources:
 
     smart_variables:
       # zone_id: "SELECT id, name FROM zones ORDER BY name ASC"
+      #=== You can use it as Array
+      # zone_name: ['Zone 1', 'Zone 2']
+      #=== or as Hash
+      # zone_type: [type_1: 'Type 1', type_2: 'Type 2']
```
Comment (ankane) [inline lib/generators/blazer/templates/config.yml]: Is this the correct YAML syntax for objects?

**Comment by ankane** [comment]
Thanks @starchow, I added some comments inline.

**Comment by starchow** [comment]
Thanks for your feedbacks @ankane 
I updated it again. Hope everything are good enough to be merged!

File: app/controllers/blazer/base_controller.rb [hunk 6]
```diff
@@ -50,6 +50,20 @@ def process_vars(statement, data_source)
       end
     end
 
+    def parse_smart_variables(var, data_source)
+      query = data_source.smart_variables[var]
+      if query.is_a? Hash
+        smart_var = query.map { |k,v| [k, v] }
```
Comment (ankane) [inline app/controllers/blazer/base_controller.rb]: I believe this needs a `reverse` to be in the format of `{value1: "Label 1", value2: "Label 2"}`.

Comment (starchow) [inline app/controllers/blazer/base_controller.rb]: Done!
I forgot it :D

**Comment by ankane** [comment]
Awesome, looks great. Thanks @starchow 👍

**Pull request merged**